app/services/moab_validation_handler.rb

```diff
@@ -73,7 +73,15 @@ def set_status_as_seen_on_disk(found_expected_version)
       return
     end
 
-    # TODO: this still isn't quite honest in the cases where checksum validation hasn't been performed
-    update_status(PreservedCopy::OK_STATUS)
+    # NOTE: subclasses which override this method should NOT perform checksum validation inside of this method!
+    # CV is expensive, and can run a while, and this method should likely be called from within a DB transaction,
+    # but CV definitely shouldn't happen inside a DB transaction.
+    if results.contains_result_code?(AuditResults::MOAB_CHECKSUM_VALID)
+      update_status(PreservedCopy::OK_STATUS)
+    elsif can_validate_checksums?
+      update_status(PreservedCopy::INVALID_CHECKSUM_STATUS)
```

Comment:
I'm not clear on lines 81-2. I see that `can_validate_checksums?` is false, and so the update on 82 won't happen. I'm wondering why that update is called at all, unless there's a case where `can_validate_checksums?` returns true, which I don't see. Can you give me some context here? Otherwise, it LGTM.

Reply:
I guess the idea is that a subclass could override that method and return true.

Reply:
Ah, yeah, I see `ChecksumValidator` does [exactly this](https://github.com/sul-dlss/preservation_catalog/blob/master/app/services/checksum_validator.rb#L63)

Reply:
Ah ok, thanks!